bin/www

```diff
@@ -1,136 +0,0 @@
-#!/usr/bin/env node
-
-/**
- * Module dependencies.
- */
-
-var app = require('../app');
-var debug = require('debug')('nodejs-regular-webapp2:server');
-var http = require('http');
-var httpProxy = require('http-proxy');
-var setAuthCookie = require('../auth');
-
-/**
- * Get port from environment and store in Express.
- */
-
-var port = normalizePort(process.env.PORT || '3000');
-app.set('port', port);
-
-/**
- * Create HTTP server.
- */
-
-var server = http.createServer(app);
-
-/**
- * Listen on provided port, on all network interfaces.
- */
-
-server.listen(port);
-server.on('error', onError);
-server.on('listening', onListening);
-
-/* Proxy websockets */
-
-var proxy = httpProxy.createProxyServer({
-  target: {
-    host: process.env.SHINY_HOST,
-    port: process.env.SHINY_PORT
-  },
-  ws: true,
-  proxyTimeout: (24 * 60 * 60 * 1000) // 1 day
-});
-
-proxy.on('error', function(err) {
-    console.log(err);
-});
-
-/*
-FIXME:
-
-Websocket requests also require that the RStudio username header be present and
-correct.
-
-However, as Passport authentication does not handle websockets we can't use
-req.user or req.session info for the username, as with HTTP requests. So instead
-we take the username from the requested hostname instead.
-
-This is imperfect, and should be replaced with an actual check for an active
-authenticated session. However, as RStudio only uses websockets for Shiny app
-previews, on paths including the current session ID and a randomly assigned
-port, websocket URLs are impossible to guess or enumerate without an active
-authenticated HTTP session, the security risk is minimal at worst.
-*/
-proxy.on('proxyReqWs', function(proxyReqWs, req, res, options) {
```

Comment:
I'm not quite sure what these are doing but I don't see them anymore in the refactored code, is that intentional?

Reply:
Yes, this is written differently in `app/proxy.js`